src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -92,20 +101,83 @@ public void doPost(HttpServletRequest request, HttpServletResponse response) thr
     float professorScore = professorSentiment.getScore();
     profLanguageService.close();
 
-    Entity professorEntity = new Entity("Professor");
-    professorEntity.setProperty("comments-professor", professorFeedback);
-    professorEntity.setProperty("score-professor", professorScore);
-    professorEntity.setProperty("perception", professorRating);
-
-    Entity commentsEntity = new Entity("Comments");
-    commentsEntity.setProperty("comments-class", classFeedback);
-    commentsEntity.setProperty("score-class", classScore);
-    commentsEntity.setProperty("perception", classRating);
+    // Gets user email.
+    UserService userService = UserServiceFactory.getUserService();
+    String userEmail = userService.getCurrentUser().getEmail();
 
+    // Check whether user has reviewer ID in system.
     DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
-    datastore.put(commentsEntity);
-    datastore.put(professorEntity);
+    Filter userFilter = new FilterPredicate("user-email", FilterOperator.EQUAL, userEmail);
+    Query userQuery = new Query("User").setFilter(userFilter);
+    List<Entity> userQueryList =
+        datastore.prepare(userQuery).asList(FetchOptions.Builder.withDefaults());
+    Entity userEntity = new Entity("User");
+
+    // User has not reviewed any rating.
+    if (userQueryList.isEmpty()) {
+      Entity newReviewer = new Entity("User");
+      newReviewer.setProperty("user-email", userEmail);
+      userEntity = newReviewer;
+    } else {
+      userEntity = userQueryList.get(0);
+    }
 
+    // Has to be added to URL.
+    currentTermKey = request.getParameter("term");
+    currentSchoolKey = request.getParameter("school");
+    Entity currentTerm = datastore.get(currentTermKey);
+    currentProfKey = currentTerm.getProperty("professor");
+
+    // Checks for term rating by reviewer.
+    Filter termReviewerFilter =
+        new FilterPredicate("reviewer-id", FilterOperator.EQUAL, userEntity.getKey());
+    Query termRatingQuery =
+        new Query("Rating Term").setAncestor(currentTermKey).setFilter(termReviewerFilter);
+    List<Entity> termRatingQueryList =
+        datastore.prepare(termRatingQuery).asList(FetchOptions.Builder.withDefaults());
+
+    // Checks for professor rating by reviewer.
```

Comment:
If I'm understanding what's below correctly you want to create a professor rating and a term rating each time a rating is submitted via this endpoint? Why is that? Where do you envision this endpoint being called from? I would expect that if you submit a review on a professor's page you only create a professor review, and same for a term. Am I misunderstanding the spec?

Reply:
Talked about this during 1:1, will implement a fix now!